tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -1002,6 +1002,159 @@ TYPED_TEST(
 	for (index_t i = 0; i < len; ++i)
 		EXPECT_NEAR(c[i] * b[i], a[i], get_epsilon<TypeParam>());
 }
+TYPED_TEST(LinalgBackendEigenAllTypesTest, SGMatrix_elementwise_division)
+{
+	const auto m = 2;
+	SGMatrix<TypeParam> A(m, m);
+	SGMatrix<TypeParam> B(m, m);
+
+	for (auto i : range(m * m))
+	{
+		A[i] = i;
+		B[i] = i + 1;
+	}
+
+	auto result = element_div(A, B);
+
+	for (auto i : range(m))
+		for (auto j : range(m))
+			EXPECT_NEAR(
+			    result(i, j), A(i, j) / B(i, j), get_epsilon<TypeParam>());
+
+	SGMatrix<TypeParam> C(m+1, m);
+	SGMatrix<TypeParam> D(m, m+1);
+	SGMatrix<TypeParam> E(m+1, m+1);
+
+	EXPECT_THROW(element_div(C, A), ShogunException);
+	EXPECT_THROW(element_div(D, A), ShogunException);
+	EXPECT_THROW(element_div(E, A), ShogunException);
+
+	EXPECT_THROW(element_div(A, C), ShogunException);
+	EXPECT_THROW(element_div(A, D), ShogunException);
+	EXPECT_THROW(element_div(A, E), ShogunException);
+}
+
+TYPED_TEST(
+    LinalgBackendEigenAllTypesTest, SGMatrix_elementwise_division_in_place)
+{
+	const auto m = 2;
+	SGMatrix<TypeParam> A(m, m);
+	SGMatrix<TypeParam> B(m, m);
+	SGMatrix<TypeParam> result(m, m);
+
+	for (auto i : range(m * m))
+	{
+		A[i] = i;
+		B[i] = i + 1;
+	}
+	
+	element_div(A, B, result);
+	for (auto i : range(m))
+		for (auto j : range(m))
+			EXPECT_NEAR(
+			    result(i, j), A(i, j) / B(i, j), get_epsilon<TypeParam>());
+	
+	SGMatrix<TypeParam> C(m+1, m);
+	SGMatrix<TypeParam> D(m, m+1);
+	SGMatrix<TypeParam> E(m+1, m+1);
+
+	EXPECT_THROW(element_div(A, B, C), ShogunException);
+	EXPECT_THROW(element_div(A, B, D), ShogunException);
+	EXPECT_THROW(element_div(A, B, E), ShogunException);
+
+	EXPECT_THROW(element_div(A, C, B), ShogunException);
+	EXPECT_THROW(element_div(A, D, B), ShogunException);
+	EXPECT_THROW(element_div(A, E, B), ShogunException);
+
+	EXPECT_THROW(element_div(C, A, B), ShogunException);
+	EXPECT_THROW(element_div(D, A, B), ShogunException);
+	EXPECT_THROW(element_div(E, A, B), ShogunException);
+}
+
+TYPED_TEST(LinalgBackendEigenAllTypesTest, SGMatrix_block_elementwise_division)
+{
+	const index_t nrows = 2;
+	const index_t ncols = 3;
+
+	SGMatrix<TypeParam> A(nrows, ncols);
+	SGMatrix<TypeParam> B(ncols, nrows);
+
+	for (auto i : range(nrows))
+		for (auto j : range(ncols))
+		{
+			A(i, j) = i * 10 + j + 1;
+			B(j, i) = i + j + 1;
+		}
+
+	const auto m = 2;
+	auto A_block = linalg::block(A, 0, 0, m, m);
+	auto B_block = linalg::block(B, 0, 0, m, m);
```

Comment:
are you going to need these block operations? I think these should be added when needed, otherwise it's just binary bloat

Reply:
so do you recommend I remove the whole thing? aka the block `element_div`?

Reply:
yes, unless you need it ofc! :)

Reply:
but can wait for @karlnapf input or maybe @vigsterkr has thoughts on this?

Reply:
do it

Reply:
we just added those for some specific cases we had